api/src/util/text_extractor.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from dataclasses import dataclass
+from io import BytesIO
+from typing import Callable
+
+import pandas
+import pptx
+from bs4 import BeautifulSoup
+from docx import Document
+from pypdf import PdfReader
+from striprtf.striprtf import rtf_to_text
+
+from src.util.file_util import open_stream
+
+logger = logging.getLogger(__name__)
+
+
+class UnsupportedTextExtractorFileType(Exception):
+    pass
+
+
+class FileTypeMismatchTextExtractorError(Exception):
+    pass
+
+
+class TextExtractorError(Exception):
+    pass
+
+
+def extract_text_from_file(
+    file_path: str, file_type: str | None = None, raise_on_error: bool = False
+) -> str | None:
+    try:
+        return TextExtractor(file_path, file_type=file_type).get_text()
+    except Exception as e:
+        if raise_on_error:
+            logger.error(e)
+            raise e
+        logger.warning(e)
+        return None
```

Comment:
I'd leave the logging out of this actually - let the calling code handle logging. In some processes we might want warnings/errors, and others we might not. Best to leave that up to the calling code.

Note that as far as our alerts are concerned, warnings/errors both should trip an alert.